lib/nodeping/client.go

```diff
@@ -235,6 +212,37 @@ func (c *Client) GetUptimesForChecks(checkIDs map[string]string, period Period)
 	return uptimes
 }
 
+func (c *Client) request(path string, v any) error {
+	req, err := http.NewRequest("GET", c.Config.BaseURL+path, nil)
+	if err != nil {
+		return fmt.Errorf("error creating request: %w", err)
+	}
+	req.Header.Set("user-agent", "sil-org/app-monitoring-archiver "+Version)
+
+	req.SetBasicAuth(c.Config.Token, "")
+
+	res, err := c.httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("error sending request: %w", err)
+	}
+	defer res.Body.Close()
+
+	body, err := io.ReadAll(res.Body)
+	if err != nil {
+		return fmt.Errorf("error reading response body: %w", err)
+	}
+
+	if res.StatusCode != 200 {
```

Comment:
Are there possibilities of any other 2XX status?

```suggestion
	if res.StatusCode != http.StatusOk {
```